report_templates/entitlements.erb

```diff
@@ -2,17 +2,28 @@
 name: Subscription - Entitlement Report
 snippet: false
 model: ReportTemplate
+template_inputs:
+- name: Days from Now
+  required: true
+  input_type: user
+  description: "Show subscriptions expiring in <= this many days.  Select 'no limit' to show all entitlements."
+  advanced: false
+  value_type: plain
+  options: "no limit\r\n7\r\n30\r\n60\r\n90\r\n120\r\n180"
 require:
 - plugin: katello
   version: 3.16.0
 -%>
-<%- report_headers 'Name', 'Organization', 'Lifecycle Environment', 'Content View', 'Host Collections', 'Virtual', 'Guest of Host', 'OS', 
-'Arch', 'Sockets', 'RAM', 'Cores', 'SLA', 'Products', 'Subscription Name', 'Subscription Type', 'Subscription Quantity', 
-'Subscription SKU', 'Subscription Contract', 'Subscription Account', 'Subscription Start', 'Subscription End', 'Subscription Guest' -%>
-<%- load_hosts(includes: [:lifecycle_environment, :operatingsystem, :architecture, :content_view, :organization, :reported_data, :subscription_facet, :pools => [:subscription]]).each_record do |host| -%>
-<%-   host.pools.each do |pool| -%>
+<%- days_from_now = input('Days from Now') -%>
+<%- days_from_now = "" if days_from_now == 'no limit' -%>
+<%- should_filter = days_from_now.present? -%>
+<%- report_headers 'Name', 'Organization', 'Lifecycle Environment', 'Content View', 'Host Collections', 'Virtual', 'Guest of Host', 'OS', 'Arch', 'Sockets', 'RAM', 'Cores', 'SLA', 'Products', 'Subscription Name', 'Subscription Type', 'Subscription Quantity', 'Subscriptions Consumed', 'Subscription SKU', 'Subscription Contract', 'Subscription Account', 'Subscription Start', 'Subscription End', 'Subscription Guest', 'Days Remaining' -%>
+<%- load_hosts(includes: [:lifecycle_environment, :operatingsystem, :architecture, :content_view, :organization, :reported_data, :subscription_facet, :pools => [:subscription]], -%>
+<%-   search: should_filter && "pools_expiring_in_days = #{days_from_now}" -%>
+<%- ).each_record do |host| -%>
+<%-   host.pools.expiring_in_days(days_from_now).sort_by { |p| p.end_date }.each do |pool| -%>
 <%-     report_row(
-          'Name': host.name,
+          'Host Name': host.name,
```

Comment:
this needs to be reflected also on line 20 in report_headers

Reply:
@ares Updated, thanks